docker_mcp/server.py

```
@@ -350,22 +390,21 @@
-        follow: bool = False,
-        # Migration-specific parameters
-        target_host_id: str = "",
-        skip_stop_source: bool = False,  # Changed: must explicitly skip stopping
-        start_target: bool = True,
-        remove_source: bool = False,
-        dry_run: bool = False
+        action: Annotated[str, Field(description="Action to perform (list, deploy, up, down, restart, build, discover, logs, migrate)")],
+        host_id: Annotated[str, Field(default="", description="Host identifier **(required for all actions)**")] = "",
+        stack_name: Annotated[str, Field(default="", description="Stack name **(used by: deploy, up, down, restart, build, logs, migrate)**")] = "",
+        compose_content: Annotated[str, Field(default="", description="Docker Compose file content **(used by: deploy)**")] = "",
+        environment: Annotated[Dict[str, str], Field(default_factory=dict, description="Environment variables **(used by: deploy, up)**")] = None,
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)